erc721 removal and inventory cleanup

0xxlegolas opened the pull request:
**### Change Log:**

**ERC721 related -**

- Remove all ERC721 code from world-v2
- Remove all references to ERC721 from Deployable, SmartCharacter, and scripts
- Remove StaticData Table, Static Data System, and all Static data reference logic
- Remove all ERC721 support tables from [mud.co](http://mud.co/)nfig
- Created an Ownership system which tracks ultimate beneficiary account ownership for both singleton and non-singleton smart objects. As well as tracking aggregate quantities of objects in the world and aggregate quantities for each individual account.
- Only items in inventory can be transferred.
- Only items in inventory ore singleton or non-singleton direct calls to deposit/withdrawal (mint/burn ownership records, call from our "interact" systems just transfer ownership
- Added ownership related tables to the mud.config
- Add new ownership tracking logic to SmartCharacter, Deployable, Inventory, and EphemeralInventory for managing account ownership tracking
- new character/deployables create will "assign" new ownership to things-in-space
- `removeCharacter`, `unanchor`, and `destoryDeployable` will "removel" ownership from things-in-space
- direct calls by the server to Inventory/Ephemeral deposit/withdrawal will also assign/remove ownership of items because this action is also like adding/removing ownership tracking of items to/from the game simulation
- any inventory calls that originate from our "interact" contracts (not a direct call) will transfer holdership from one inventory to another, and transfer ownership from one inventory owner to another 

**SmartCharacter related -**

- Update param naming convention in SmartCharacter to use `smartObjectId` for consistency
- Adjusted Characters Table (remove `characterAddress` since all smartobject-to-account tracking can now be handled the OwnershipSystem)
- Add `removeCharacter` logic to SmartCharacterSystem
- error reporting updates

**Deployable related -**

- Adjusted the `unanchor` and `destroyDeployable` cases to also remove ownership linkage 
- Adjusted `Deployable.createDeploayble` to only allow the NULL state, Not UNANCHORED . This was a previous workaround, but this is unsafe , when you re-anchor you should call anchor not create.
- function name shortening where possible
- error reporting updates
- Inventory version updating when `unanchor` or `destroy` called
- Inventory used capacity reset when `unanchor` or `destroy` called
- smart gate unlink called when `unanchor` or `destroy` called
- `owner` added to anchor params, to track ownership upon re-anchoring

**Fuel related -** 

- add strict parameter range restrictions to prevent any overflow/underflow cases in Fuel logic

**Location related -** 
- remove `Coord` and `WorldLocation` types in favor of using the LocationData table struct itself

**EntityRecords related -**

- Add a Tenant table for tracking the acceptable tenant value for this World for EntityRecords
- Update EntityRecord with `tenantId` for full game data tracking
- Add EntityRecord sanity checks to Inventory/Ephemeral Item create flows, as well as to Deployable creation (in SmartAssembly) and Smart Character creation.
- function name shortening where possible
- make EntityRecord fields consistent everywhere in the whole codebase

**AccessSystem related -**

- Made AccessSystem consistent in terms of param and function naming (use smartObjectId everywhere).
- Added several Access logic rule for configuration
- Update AccessSystem to use new `Ownership.owner()` getter for `isOwner` checks
- Re-implement role access rules for Inventory/EphemeralInteract in AccessSystem
- adjusted the access for `EntitySystem.deleteClass` to NO ACCESS, deleting a class currently leaves a bunch of orphaned object data with unhandled data corruption, so we prohibit it for now.
- Implement Ownership access rule with scope enforcement for functions which can have ephemeral objects as an input - by using ObjectByEphemeral lookups
- Added special case access for entity record - for SmartCharacter - only admin can set name, and set metadata
- Add EntityRecord parameter sanity checks to Inventory, EphemeralInventory and Deployable create functions
- Add `CallAccess` rules to Access System (standardized way for system-to-system call access)

**Inventory related -**

- Updated Inventory and EphemeralInventory item ID generation logic (I.e. only create classId EntityRecords and Entity entries when they don’t exist)
- change `inventoryItemId` naming to `itemObjectId` for consistency in `mud.config` and elsewhere
- Added `NULL` to  the deployable state checks for inventory actions (this will allow non-deployables to also have inventory, since no created Deployable has state NULL) this is to accommodate adding inventory to rifts which are Assemblies (but not Deployable), and will have inventory
- Reworked Inventory/Interact role logic - no more role assignment in the capacity setting logic. Now the SmartObject Owner (as per the OwnershipSystem logic) can set InventoryInteract roles for who can interact with functionality.
- Updated the role setting logic to use these InventoryInteract specific roles and the functions themselves now check those specific roles for access.
- Moved `setEphemeralCapacity` to the InventorySystem.. since this is global capacity setter at the SmartObject level, whereas all other EphemeralInventory functionality is per ephemeral inventory
- Reworked Inventory/Ephemeral deposit flows to better match the withdrawal logic flows
- Rename InventoryInteract to `EphemeralInteract` (as in... this contract is for interacting with Ephemeral inventories from an Object's primary inventory)
- Created an new contract called `InventoryInteract` (for explicitly interacting with other smart object inventories from a primary inventory)
- `InventoryInteract` has exactly same pattern as `EphemeralInteract` except this handles transfers from one smart object inventory to another object inventory (instead of a smart object inventory to an associated ephemeral inventory)
- Added another transfer tracking table called `InventoryItemTransfer` to track object-to-object inventory transfers
- Renamed `ItemTransfer` table to `EphemeralItemTransfer`
- function name shortening where possible
- error reporting updates
- decoupled item data versioning from deployable state, it now only relies on the inventory `version` for stale data tracking

**SmartGate related -**
- Update link unlink logic
- Add auto-unlinking in Deployable when state changes to `UNANCHORED` or `DESTROYED`

**Parameter/Types related -**

- Rename CreateAndAnchorDeployableParams to `CreateAndAnchorParams`
- Remove `SmartObjectData` type from CreateAndAnchorParams
- rename InventoryItem to `CreateInventoryItemParams` and TransferItem to `InventoryItemParams`.
- Use `InventoryItemParams` consistently everywhere (besides the create functions)
- remove the `owner` param from both CreateInventoryItemParams and InventoryItemParams
- Change name of EntityRecordData to `EntityRecordParams` and EntityMetadata to `EntityMetadataParams`, to avoid conflict with the similarly named Table struct and consistency.

**General -**

- Remove SOFAccess Utils & Remove InventoryUtils
- Remove `common-constants` dependency in world-v2 package.json
- Rebase with the latest develop branch and init pattern changes
- Rename ConfigSof to `Config`, since this script configures the EveSystem setup and World access also (not just SOF)
- Add the `OwnershipSystem` to EveSystem for class initializations
- Added `Tenant` setting to the `Config` script and the .env
- Update `Config` script to reflect the latest changes
- Configure Inventory/EphemeralInventory/Ownership CallAccess in Config script
- Update naming and access matrix in EveSystem
- Add role based access control configurations for the OwnershipSystem and Inventory/EpemeralInteract updates to EveSystem
- `tenetId` added to classId generation logic, to prevent cross-chain asset usage
- `EntityRecord` creation added to the class `initialize()` flow - this will be updated in the future to be more robust and lees reliant on the game data directly

fc-tezza committed:
mid-develop commit

fc-tezza committed:
Merge branch 'develop' into ccp-tezza/erc721-removal

fc-tezza committed:
all code changes compiling

fc-tezza committed:
fix naming conflicts

fc-tezza committed:
Merge branch 'develop' into ccp-tezza/erc721-removal-and-inventory-cleanup

fc-tezza committed:
all post merge updates completed including access configuration

fc-tezza committed:
inventory test nearly done, stuck on test deploy

fc-tezza committed:
keep previous interface structure for SOF

fc-tezza committed:
move SOF errors into systems for error reporting and update SOF tests to match updates

fc-tezza committed:
commit to checkout new branch from here

fc-tezza committed:
inv test mid stream

fc-tezza committed:
inventory unit tests complete

fc-tezza committed:
ephemeral unit tests complete, update classId generation method to include tenantId

fc-tezza committed:
classId fix in EveSystem

fc-tezza committed:
EphemeralInteract setup

fc-tezza committed:
add SSU bootstrapping to tests, add classId entity record generation to EveSystem.initialize()

fc-tezza committed:
inventory and ephemeral interact unit tests complete

fc-tezza committed:
ownership uint tests complete

fc-tezza committed:
deployable unit tests complete, fix for gloabl pause logic

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/EveSystem.sol:
```diff
@@ -510,11 +614,16 @@ contract EveSystem is IEveSystem, SmartObjectFramework {
+    uint256 classId = uint256(keccak256(abi.encodePacked(Tenant.get(), typeId)));
     entitySystem.scopedRegisterClass(classId, _callMsgSender(1), systemIds);
-
+    entityRecordSystem.createRecord(classId, EntityRecordParams({
```
what is the issue in creating a entity record when a ssu creation flow starts from the game ? 
Why should this be part of deployment process ?

fc-tezza replied:
This is a stop gap solution until we settle exactly how we are going to get class EntityRecord data. We could use the inventory pattern of generation just-in-time. 

But I also viewed that as a temporary (flawed) solution since it is not particularly robust. it relies on the fact that the game servers will give us accurate data in every call, which seems pretty brittle.

Ultimately, I think the class data pre-population flow combined with some decoupling of IDs from hard reliance on game data is probably the best long term solution.

0xxlegolas replied:
> But I also viewed that as a temporary (flawed) solution since it is not particularly robust. it relies on the fact that the game servers will give us accurate data in every call, which seems pretty brittle.

The same issue could happen during deployment too, right? 

> Ultimately, I think the class data pre-population flow combined with some decoupling of IDs from hard reliance on game data is probably the best long term solution.
agreed

fc-tezza replied:
> The same issue could happen during deployment too, right?

Yes but at least there it is only one time and easily track-able. During on-going operational calls would be far harder to monitor and find issues. Regardless, we should update this in the near-ish future.

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/EveSystem.sol:
```diff
@@ -108,89 +121,135 @@ contract EveSystem is IEveSystem, SmartObjectFramework {
+    ];
 
-    for (uint256 i = 0; i < onlyAdminSelectors.length; i++) {
+    for (uint256 i = 0; i < entityRecordOnlyAdminForCharactersOtherwiseAlsoOwnerSelectors.length; i++) {
```
`entityRecordOnlyAdminForCharactersOtherwiseAlsoOwnerSelectors ` the variable name is too long. Can we stick to length <25 for function names and variable names ?

fc-tezza replied:
yeah, it's hard to be descriptive with shorter versions, but I agree these names are too long!

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/deployable/DeployableSystem.sol:
```diff
@@ -197,13 +210,22 @@ contract DeployableSystem is SmartObjectFramework {
+    if (currentOwner != address(0) && currentOwner != owner) {
+      ownershipSystem.annulFromAccount(smartObjectId, currentOwner);
+      ownershipSystem.ascribeToAccount(smartObjectId, owner);
+    } else if (currentOwner == address(0)) {
```
might be best if the cases are swapped as the most used case is `currentOwner == address(0)` so it does not have to go to else in most cases

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -31,35 +31,47 @@ export default defineWorld({
-         * Used to store the assembly of a smart object
-         */
-        SmartAssembly: {
+        InventoryByItem: {
```
```suggestion
        InventoryItem: {
```

fc-tezza replied:
these are different tables, storing different things, we already have an `InventoryItem` table

fc-tezza replied:
I'm using the CharactersByAccount pattern for naming for reverse lookup tables

0xxlegolas replied:
got it

0xxlegolas replied:
Then `ItemToInventoryMap` ?

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -31,35 +31,47 @@ export default defineWorld({
+          },
+          key: ["itemObjectId"],
+        },
+        OwnershipByObject: {
```
```suggestion
        Owner: {
```

fc-tezza replied:
I'm using the CharactersByAccount pattern for naming for reverse lookup tables

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -72,134 +84,42 @@ export default defineWorld({
-         * ERC721 PUPPET MODULE *
-         ************************/
-        Balances: {
+        CharactersByAccount: {
```
```suggestion
        Characters: {
```

fc-tezza replied:
We already have a `Characters` table.. `CharactersByAccount `is the reverse lookup table

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -272,33 +181,51 @@ export default defineWorld({
+        /**
+         * Used to signal the transfer details when a item is exchanged between its primary inventory and another smart object's primary inventory
+         */
+        ObjectItemTransfer: {
```
```suggestion
        InventoryItemTransfer: {
```

fc-tezza replied:
Sounds good

0xxlegolas replied:
Does this table also track ephemeral to ephemeral transfer of items ?

0xxlegolas replied:
I saw itemTransfer table is also doing something similar so can we rename as `IntraInventoryTransfer`

fc-tezza replied:
I've opted for EphemeralItemTransfer (for EphemeralInteract) and InventoryItemTransfer (for InventoryInteract)

0xxlegolas replied:
This sounds good but I noticed 

`InventoryItemTransfer` does inventory to inventory transfer 
`EphemeralItemTransfer` does inventory to ephemeral, ephemeral to inventory and ephemeral to ephemeral.

fc-tezza replied:
yeah, the `Ephemeral` naming just signifies that it handles everything to do with Ephemeral interactions.

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -372,7 +303,6 @@ export default defineWorld({
           key: ["smartObjectId"],
         },
-
         SmartGateLink: {
```
Made couple of comments on the table names to keep it simple and intuitive. Let me know if it works

fc-tezza committed:
fuel tests mid-stream

0xxlegolas commented on mud-contracts/smart-object-framework-v2/src/namespaces/sofaccess/systems/sof-access-system/SOFAccessSystem.sol:
```diff
@@ -257,6 +255,21 @@ contract SOFAccessSystem is ISOFAccessSystem, SmartObjectFramework {
+   * @param targetCallData The calldata of the target function
+   * @dev Handles calls to EntitySystem.deleteClass, currently there are too many un-resolved data dependencies to allow for any access to this function
+   */
+  function noAllowances(uint256 entityId, bytes memory targetCallData) public view {
```
Just curious—if some functions aren’t meant to be accessed anymore, is there a reason to keep them around? Wondering if it makes sense to keep obsolete code.

fc-tezza replied:
`deleteClass` is prohibited for now until we can find a graceful solution for data dependency issues

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -309,49 +236,54 @@ export default defineWorld({
-          key: ["smartObjectId", "inventoryItemId", "ephemeralInvOwner"],
+          key: ["smartObjectId", "ephemeralOwner", "itemObjectId"],
+        },
+        ObjectByEphemeral: {
```
What’s the purpose of the ObjectByEphemeral table? I’m having trouble understanding its use—since the key is ephemeralSmartObjectId, it seems like the user would already need both smartObjectId and ephemeralOwner to construct it. If they already have those values, is there really a need to look it up in a table?

fc-tezza replied:
Ephemeral inventories are assigned an ephemeral object id which is generated combining their lookup keys (it is a unique ID `keccack256(inventorySmartObjectId, ephemeralOwner)`

This `ephemeralObjectId` populated in `InventoryByItem.set(itemObjectId, ephemeralInventoryId)` when an item is assigned to an ephemeral inventory.

For lookup purposes we also then have `ObjectByEphemeral` which is another reverse lookup table which allows us to input the `ephermalObjectId `to lookup the associated primary inventory and the owner.

With this we can find both the primary inventory smart object Id (the ID of the SSU or other deployable) and the owner of the ephemeral inventory holding that item without having to know those pieces of dat beforehand.. 

Under this pattern, we only need to input the `itemObjectId` and we can find all of this ownership data.

0xxlegolas replied:
The naming got me confused, Does this work 

```
/**
         * Look up table to find the associated inventory smart object for an ephemeral inventory
         */
        InventoryByEphemeral: {
          schema: {
            ephemeralSmartObjectId: "uint256",
            exists: "bool",
            inventorySmartObjectId: "uint256", // parent container ID
            inventoryOwner: "address", 
          },
          key: ["ephemeralSmartObjectId"],
        },
```

fc-tezza replied:
Yeah, sounds good

0xxlegolas commented on mud-contracts/world-v2/mud.config.ts:
```diff
@@ -309,49 +236,54 @@ export default defineWorld({
+         * Used to signal the transfer details when a item is exchanged between it's primary inventory and an associated ephemeral inventory or two ehpemeral inventories
          */
-        ItemTransferOffchain: {
+        ItemTransfer: {
```
Now that we have two types of transfers—between different inventories and within the same one—would it make sense to rename them to `IntraInventoryEphemeralTransfer` and `InterInventoryEphemeralTransfer`? Might help make things more intuitive for integrations

0xxlegolas replied:
Sorry, noticed ephemeral to ephemeral transfer is also happening, so we can leave it as it is.

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/deployable/DeployableSystem.sol:
```diff
@@ -106,20 +105,22 @@ contract DeployableSystem is SmartObjectFramework {
-      staticDataSystem.setCid(smartObjectId, smartObjectData.tokenURI);
+    // TODO: the following is a candidate for hook logic
+    // set the initial inventory data version to 1
+    if (Inventory.getVersion(smartObjectId) == 0) {
```
Shouldn't we revert if its not 0 ?

fc-tezza replied:
Currently there is no possibility for getting this far in the code with a non-zero version.

There are only two places that version gets bumped - when a deployable is unanchored or destroyed. In both case the state of the deployable will be not-NULL and can never go back to the NULL state.

Because of this the earlier NULL state check would prohibit us from ever getting this far in the code with a non-zero  versioned inventory, because any such deployable would fail the NULL check.

so actually instead, I guess we don't even need a zero check here.. we can just set to 1 regardless because at this point it will ALWAYS be a zero version.

fc-tezza replied:
But this did remind me that not all deployables necessarily have an inventory, so I've updated the code to check and see if the current deployable smartObject has Inventory scoped/tagged to its class and only set a version if an inventory is possible.

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/deployable/DeployableSystem.sol:
```diff
@@ -197,13 +210,22 @@ contract DeployableSystem is SmartObjectFramework {
     LocationData memory locationData
   ) public onlyActive context access(smartObjectId) scope(smartObjectId) {
     State previousState = DeployableState.getCurrentState(smartObjectId);
     if (previousState != State.UNANCHORED) {
```
Should we also include || previousState == State.UNANCHORED here? Since a deployable can go from unanchored to anchored ? 

May be this was the reason the above condition was added in createDeployable

fc-tezza replied:
We are checking the UNANCHORED requirement on line 216, just below your comment here.

fc-tezza committed:
fuel unit tests completed, and fuzzing re-added for fuel calculations

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/inventory/EphemeralInventorySystem.sol:
```diff
@@ -1,346 +1,366 @@
     {
       State currentState = DeployableState.getCurrentState(smartObjectId);
-      if (currentState != State.ONLINE) {
+      if (!(currentState == State.NULL || currentState == State.ONLINE)) {
```
why `NULL` state is included ?

0xxlegolas replied:
I saw the comment its for the inventories not attached with deployables

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/smart-assembly/SmartAssemblySystem.sol:
```diff
@@ -1,82 +1,107 @@
-    uint256 smartAssemblyEnumId = SmartAssembly.getSmartAssemblyId(smartObjectId);
-    smartAssemblyEnumId = smartAssemblyEnumId + 1;
+    uint256 assemblyEnumId = SmartAssembly.getAssemblyId(smartObjectId);
+    assemblyEnumId = assemblyEnumId + 1;
```
The enum ID should probably be predefined—like SSU = 0, ST = 1, SG = 2. If we’re incrementing it with every createAssembly, which now gets called during every createAndAnchor, then we’ll end up incrementing the ID for each SSU. Also, I noticed that createSmartAssembly is only being used for deployables, not for gates or turrets

fc-tezza replied:
> The enum ID should probably be predefined—like SSU = 0, ST = 1, SG = 2. If we’re incrementing it with every createAssembly, which now gets called during every createAndAnchor, then we’ll end up incrementing the ID for each SSU.

I think the current logic just assigns 1 every time to each `smartObjectId`, unless that `smartObjectId `has had its assemblyType set before (which shouldn't/doesn't happen in our code).

So I don't think this will do as you mention, but I do think we should review what exactly we want this to do.

If we want to make it static and want it independently trackable from `typeId` data, we could just as easily `unit256(keccak256(assemblyType))` for its uint id. Was there a reason we want it to be an enum?

> Also, I noticed that createSmartAssembly is only being used for deployables, not for gates or turrets

gates and turrets ARE deployables, so they are covered.

0xxlegolas replied:
The best way could be defining  like this  :  https://github.com/projectawakening/world-chain-contracts/blob/develop/mud-contracts/world/mud.config.ts#L69

fc-tezza replied:
As we discussed, I've just removed the enum number tracking completely since we don't use it anywhere currently.

0xxlegolas committed:
- better-names-less-pain
- updated function names for clarity, replacing 'ascribe' with 'assign' and 'annul' with 'remove'.
- adjusted related tests and imports to reflect these changes, ensuring consistency in the ownership and inventory systems.

0xxlegolas committed:
add build files

fc-tezza committed:
entityrecord and location uint tests completed

fc-tezza committed:
update deployable inventory checks and tests

fc-tezza committed:
smart assembly uint tests completed

fc-tezza committed:
smart character unit tests completed

0xxlegolas committed:
Add gates online checks and unlinking logic in SmartGateSystem

- Introduced error handling for gates not being online.
- Enhanced the unlinking process to handle destroyed gates appropriately.

0xxlegolas committed:
deouple ownership management

- Added InventoryOwnershipSystem to manage inventory ownership assignments and removals.
- Adjusted tests to validate the new inventory ownership logic and error handling.

fc-tezza committed:
smart gate, smart storage unit, smart turret unit tests complete

fc-tezza committed:
update ItemTransfer table naming

fc-tezza committed:
linter format fix

fc-tezza committed:
import clean up

fc-tezza committed:
add killmail to v2, clean up EveSystem imports

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/smart-gate/SmartGateSystem.sol:
```diff
@@ -52,26 +54,36 @@ contract SmartGateSystem is SmartObjectFramework {
     }
 
-    //Create a 2 way link between the gates
+    // Create a 2 way link between the gates
```
delete links before unlink is missing 

```
    //Delete the existing records for the source and destination gate before creating a new link to avoid replacing the record
    //The invalid records are not deleted during unlink because the external services are subscribed to the unlink events. If the record is deleted then the external services will not be able to notify the game
    _deleteExistingLink(sourceGateId);
    _deleteExistingLink(destinationGateId);
```

fc-tezza replied:
Links are now auto-deleted in the Deployable logic [here](https://github.com/projectawakening/world-chain-contracts/blob/95f1df71d7c6f24855601e0b5fa12da28c4fee54/mud-contracts/world-v2/src/namespaces/evefrontier/systems/deployable/DeployableSystem.sol#L172) and [here](https://github.com/projectawakening/world-chain-contracts/blob/95f1df71d7c6f24855601e0b5fa12da28c4fee54/mud-contracts/world-v2/src/namespaces/evefrontier/systems/deployable/DeployableSystem.sol#L281), or manually [here](https://github.com/projectawakening/world-chain-contracts/blob/95f1df71d7c6f24855601e0b5fa12da28c4fee54/mud-contracts/world-v2/src/namespaces/evefrontier/systems/smart-gate/SmartGateSystem.sol#L96). 
There is no need to have the delete logic in this function now.. if either gate has a link still, [we will revert before make it this far](https://github.com/projectawakening/world-chain-contracts/blob/95f1df71d7c6f24855601e0b5fa12da28c4fee54/mud-contracts/world-v2/src/namespaces/evefrontier/systems/smart-gate/SmartGateSystem.sol#L74) in the logic. And if there were old link data still, we will just overwrite any old link data with new links [here](https://github.com/projectawakening/world-chain-contracts/blob/95f1df71d7c6f24855601e0b5fa12da28c4fee54/mud-contracts/world-v2/src/namespaces/evefrontier/systems/smart-gate/SmartGateSystem.sol#L87).

fc-tezza replied:
The unlink event will still be triggered.. but it will come from the deployable logic now when appropriate state change occurs.

fc-tezza replied:
I've added this back in to satisfy our integration needs. We should explore a better pattern for the integration data needs.

0xxlegolas commented on mud-contracts/world-v2/src/namespaces/evefrontier/systems/smart-gate/SmartGateSystem.sol:
```diff
@@ -113,19 +127,19 @@ contract SmartGateSystem is SmartObjectFramework {
-
-    if (destinationGateState != State.ONLINE) {
-      revert DeployableSystem.Deployable_IncorrectState(destinationGateId, destinationGateState);
+    // Check if the gates are online
```
Do we really need multiple if-else checks here? Isn’t the main goal just to show which gate isn’t online?

fc-tezza replied:
I just thought it is useful information to know which gate is not online and blocking the call for error reporting, 

We could collapse this into `source-not-online || destination-not-online` but then we lose the ability to report which gate is the offender in the error.

0xxlegolas replied:
Based on the previous code, if we do something like:
```
if (sourceGateState != State.ONLINE) {  
   // revert: source gate is not online  
}  
  
if (destinationGateState != State.ONLINE) {  
   // revert: destination gate is not online  
}  
```
this should indicate which gate isn’t online, right?

fc-tezza replied:
But that doesn't cover the case of reporting that BOTH of them are offline, which is why it is structured how it is currently.

0xxlegolas committed:
remove inventory management tests from ownership tests

0xxlegolas committed:
linter fixes

fc-tezza committed:
Merge branch 'develop' into ccp-tezza/erc721-removal-and-inventory-cleanup

fc-tezza committed:
add data clearing logic back into SmartGate.linkGates for the interfacing flow

fc-tezza committed:
rebuild pnpm-lock.yaml

0xxlegolas committed:
Merge branch 'ccp-tezza/erc721-removal-and-inventory-cleanup' of github.com:projectawakening/world-chain-contracts into improvement/refactor-and-fixes

0xxlegolas committed:
Merge branch 'ccp-tezza/erc721-removal-and-inventory-cleanup' of github.com:projectawakening/world-chain-contracts into improvement/refactor-and-fixes

0xxlegolas committed:
chore: lint

0xxlegolas committed:
remove unlinkInvalidGates

0xxlegolas committed:
chore: change naming

0xxlegolas committed:
remove duplicates

fc-tezza committed:
ObjectByEphermal to InventoryByEphemeral name change, test table import cleanup

fc-tezza committed:
Merge remote-tracking branch 'origin/improvement/refactor-and-fixes' into ccp-tezza/erc721-removal-and-inventory-cleanup

fc-tezza committed:
rename assignOwnerToInveentory/removeOwnerFromInventory to assignItemToInventory/removeItemFromInventory

fc-tezza committed:
re-add inventory and ephemeral interaction testing to EntityRecordTest

fc-tezza committed:
pass format linting

fc-tezza committed:
Merge branch 'ccp-tezza/world-v2-killmail' into ccp-tezza/erc721-removal-and-inventory-cleanup

fc-tezza committed:
update EveSystem imports and interface, update Config to include ownership and killmail access config, update ownership tests with access enforcement on

0xxlegolas committed:
end end tests v2 (#499)

0xxlegolas committed:
fix tests (#501)
